docs/contributing.adoc

```diff
@@ -265,4 +265,4 @@ $ nix-shell --pure tests -A run.alacritty-empty-settings
 However, those invocations will impurely source the system’s nixpkgs, and may cause failures. To run against the nixpkgs from the flake.lock, use instead e.g.
 
 [source,console]
-$ nix develop --ignore-environment .#tests.all
+$ nix develop --ignore-environment .#all
```

Comment:
I'm actually not a fan of this lack of namespacing. I think it would make more sense to resuscitate https://github.com/nix-community/home-manager/pull/3082 and expose the tests as `legacyPackages` instead of `packages`, so you could do `nix develop .#tests.run.all` (we can't do that with `packages` without angering `nix flake check`, cf. https://github.com/nix-community/home-manager/pull/3905).

Reply:
Is there a particular reason we use `devShells` instead of `checks`?

Reply:
I think the problem is that `checks` expect to be built, while our test suite (based on [nmt](https://git.sr.ht/~rycee/nmt)) must be run in a shell (hence `nix develop`).

Reply:
...aren't most common Nix builders written to use shells? `stdenv.mkDerivation` is literally a Nix framework to write a shell-script builder, which is then invoked as the build function.

TBH, I don't even see anything in `nmt` that would likely be problematic -- just how the test framework creates dummy packages, but that's in `/tests/stubs.nix`, not `nmt` itself, where pulling out that function into a separate top-level function (whether under `lib` in `flake.nix` or as itself in `default.nix`

Reply:
Yeah, disregard that. We can actually just use `tests.build.all` once the space leak is fixed (https://gitlab.com/rycee/nmt/-/merge_requests/6)

Reply:
In that case, is this good to merge as-is for now?